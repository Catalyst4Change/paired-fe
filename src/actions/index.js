export const setLoading = bool => ({
  type: 'SET_LOADING',
  bool
});

export const setError = error => ({
  type: 'SET_ERROR',
  error
});

export const setUser = user => ({
  type: 'SET_USER',
  user
});