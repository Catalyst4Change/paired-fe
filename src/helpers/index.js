export const determineDisplayTime = time => {
  switch (time) {
    case 'morning':
      return '8:00 - 8:50 a.m.';
    case 'lunch':
      return '12:00 - 12:50 p.m.';
    case 'afternoon':
      return '4:10 - 5:00 p.m.';
    default:
      return;
  }
};

export const getDatesToDisplay = (num) => {
         let dates = [];
         let counter = 1;
         while (dates.length < num) {
           let nextDate = new Date();
           nextDate.setDate(nextDate.getDate() + counter);
           if (nextDate.getDay() !== 0 && nextDate.getDay() !== 6) {
             dates.push(nextDate.toString().slice(0, 15));
           }
           counter += 1;
         }
         return [
           ...dates,
         ];
       };