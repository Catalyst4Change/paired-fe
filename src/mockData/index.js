export const mockUser = {
  name: "Hillary",
  program: "FE",
  module: 4,
  id: "5caa7eebfdebb8348e53a48e"
}

export const mockSchedule = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    pairee: null,
    date: "Mon Apr 15 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a490",
    notes: null
  },
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  }
];

export const mockAvailPairings = [
  {
    pairer: {
      name: "Tiffany",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48d",
      pronouns: "she/her",
      skills: [
        "react",
        "testing"
      ],
      slack: "@tiffany",
    },
    date: "Thu Apr 18 2019",
    time: "morning",
    id: "5caa7eebfdebb8348e53a4a4",
  },
  {
    pairer: {
      name: "Tiffany",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48d",
      pronouns: "she/her",
      skills: [
        "react",
        "testing"
      ],
      slack: "@tiffany",
    },
    date: "Thu Apr 18 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a4a5",
  }
]; 

export const mockPairingToDeleteFromSched = {
  pairer: {
    name: 'Hillary',
    module: 4,
    program: 'FE',
    id: '5caa7eebfdebb8348e53a48e',
    pronouns: 'she/her',
    slack: '@hillstew'
  },
  pairee: null,
  date: 'Mon Apr 15 2019',
  time: 'lunch',
  id: '5caa7eebfdebb8348e53a490',
  notes: null
};

export const mockScheduleAfterDelete = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  }
]; 

export const mockPairingToAddToSched = {
  pairer: {
    name: 'Hillary',
    module: 4,
    program: 'FE',
    id: '5caa7eebfdebb8348e53a48e',
    pronouns: 'she/her',
    slack: '@hillstew'
  },
  pairee: null,
  date: 'Thu Apr 18 2019',
  time: 'morning',
  id: '5caa7eebfdebb8348e56f000',
  notes: null
};

export const mockScheduleAfterAdd = [
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    pairee: null,
    date: "Mon Apr 15 2019",
    time: "lunch",
    id: "5caa7eebfdebb8348e53a490",
    notes: null
  },
  {
    pairer: {
      name: "Hillary",
      module: 4,
      program: "FE",
      id: "5caa7eebfdebb8348e53a48e",
      pronouns: "she/her",
      slack: "@hillstew"
    },
    paireee: {
      name: "Jeo",
      module: 4,
      program: "FE",
      pronouns: "he/him",
      slack: "@jeo",
    },
    date: "Wed Apr 17 2019",
    time: "afternoon",
    id: "5caa7eebfdebb8348e53a49d",
    notes: null,
  },
  mockPairingToAddToSched
];

export const mockQueryFromgetAvailablePairings = {
         query: `{
    getAvailablePairings(filter: { program: "BE", module: 3, date: "Wed Apr 17 2019" }) {
      pairer {
        name
        module
        program
        id
        pronouns
        skills
        slack
      }
      date
      time
      id
    }
  }`
       };

export const mockQueryFromgetUser = {
         query: `{
  getUser(name: "Hillary") {
    name
    program
    module
    id
  }
}`
       };

export const mockQueryFromgetUserPairings = {
         query: `{
    getUserPairings(
      id: "5h5h5h5h5h5h5"
    ) {
      pairer {
        name
        module
        program
        id
        pronouns
        slack
      }
      pairee {
        name
        module
        program
        pronouns
        slack
      }
      date
      time
      id
      notes
    }
  }`
       };

export const mockQueryFromupdatePairing = {
         query: `mutation {
    updatePairing(pairing: {
        id: "6t6t6t"
        pairee: "7b7b7b"
        notes: "Mythical Creatures"
    }) {
      pairer {
        name
        module
        program
        id
        pronouns
        slack
      }
      pairee {
        name
        module
        program
        id
        pronouns
        slack
      }
      date
      time
      id
      notes
    }
  }`
       };

export const mockQueryFromdeletePairing = {
         query: `mutation {
    deletePairing(id: "45dtl") {
      date
    }
  }`
       };