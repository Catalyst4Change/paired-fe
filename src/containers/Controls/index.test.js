import React from 'react';
import { shallow } from 'enzyme';
import { Controls, mapStateToProps, mapDispatchToProps } from './index';
import { mockAvailPairings } from '../../mockData';
import { setError } from '../../actions';
import * as helpers from '../../helpers';

describe('Controls', () => {
  describe('Controls component', () => {
    let wrapper;
    const mockSetError = jest.fn();
    const mockSetAvailPairings = jest.fn();
    const mockProps = {
      setError: mockSetError,
      setAvailPairings: mockSetAvailPairings,
      availPairings: mockAvailPairings
    }

    beforeEach(() => {
      wrapper = shallow(<Controls {...mockProps} />);
      helpers.getDatesToDisplay = jest.fn(() => [
        'Mon Apr 15 2019',
        'Tue Apr 16 2019',
        'Wed Apr 17 2019',
        'Thu Apr 18 2019',
        'Fri Apr 19 2019'
      ]);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with availPairings', () => {
      const expected = { availPairings: mockAvailPairings };
      const mockState = {
        availPairings: mockAvailPairings,
        isLoading: false,
        hasError: '',
        user: {}
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn()
    it('should call dispatch with setError when setErro is called', () => {
      const mockError = 'Error loading';
      const actionToDispatch = setError(mockError);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setError(mockError);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});