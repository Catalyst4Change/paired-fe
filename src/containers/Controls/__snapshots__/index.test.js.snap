// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Controls component should match the snapshot 1`] = `
<div
  className="Controls"
>
  <h2>
    Book a Pairing
  </h2>
  <section
    className="Controls--section"
  >
    <Dropdown
      handleChange={[Function]}
      label="Program"
      options={
        Array [
          "FE",
          "BE",
        ]
      }
    />
    <Dropdown
      handleChange={[Function]}
      label="Module"
      options={
        Array [
          "1",
          "2",
          "3",
          "4",
        ]
      }
    />
    <Dropdown
      handleChange={[Function]}
      label="Date"
      options={
        Array [
          "Thu Apr 11 2019",
          "Fri Apr 12 2019",
          "Mon Apr 15 2019",
          "Tue Apr 16 2019",
          "Wed Apr 17 2019",
        ]
      }
    />
  </section>
  <button
    className="Controls--avail-btn"
    disabled={true}
    onClick={[Function]}
  >
    Show Available Pairings
  </button>
  <Pairings
    openPairings={
      Array [
        Object {
          "date": "Thu Apr 18 2019",
          "id": "5caa7eebfdebb8348e53a4a4",
          "pairer": Object {
            "id": "5caa7eebfdebb8348e53a48d",
            "image": "https://avatars1.githubusercontent.com/u/40586291?v=",
            "module": 4,
            "name": "Tiffany",
            "program": "FE",
            "pronouns": "she/her",
            "skills": Array [
              "react",
              "testing",
            ],
            "slack": "@tiffany",
          },
          "time": "morning",
        },
        Object {
          "date": "Thu Apr 18 2019",
          "id": "5caa7eebfdebb8348e53a4a5",
          "pairer": Object {
            "id": "5caa7eebfdebb8348e53a48d",
            "image": "https://avatars1.githubusercontent.com/u/40586291?v=",
            "module": 4,
            "name": "Tiffany",
            "program": "FE",
            "pronouns": "she/her",
            "skills": Array [
              "react",
              "testing",
            ],
            "slack": "@tiffany",
          },
          "time": "lunch",
        },
        Object {
          "date": "Thu Apr 18 2019",
          "id": "5caa7eebfdebb8348e58076",
          "pairer": Object {
            "id": "5caa7eebfdebb8348e53a48f",
            "image": "https://avatars1.githubusercontent.com/u/40586291?v=",
            "name": "Jeo",
            "program": "FE",
            "pronouns": "he/him",
            "skills": Array [
              "graphql",
              "testing",
            ],
            "slack": "@jeo",
          },
          "time": "lunch",
        },
      ]
    }
  />
</div>
`;
